Update styles to support using attributes for timeline elements

walterg2 opened the pull request:
## Problem

Currently, the style sheet in place has a few places where it is looking for the `timeline-*` tags instead of relying on CSS classes. This causes users that want to keep their HTML valid issues as they need to duplicate the styles for `angular-timeline`.
### Changes
- Updated packages to work with Node 6.x (previous versions of packages would not compile correctly)
- Updated SCSS file to support using attributes
  - Cleaned up the SCSS to take advantage of nested classes
  - Swapped the `img` style to be nested inside the `timeline` class so it would not collide with other styles set up by the user
- Added an example that renders the timeline using only HTML `data` attributes
#### Notes
- I did not include updated `ngdocs` due to it not correctly interpolating the links
- Package version was not bumped for this. While I believe this would fall under a `revision` classification, I felt it best to leave it up to you on how you wanted to update the version

Please let me know if you'd like any changes to what I've done here.

rpocklin commented:
Thanks for the PR.  Can I ask what browsers you are supporting in your case?  From memory IE7 or IE8 don't support custom HTML tags but generally speaking, angular.js directives use custom tags in this way.

walterg2 commented:
We support IE9+, Latest ESR Firefox, and Chrome mainly. This change was more due to the fact that when applying the directives via attributes (which they fully support), some of the styles were not being applied because they were directly tied to the custom tag. I mainly saw this on `timeline-badge` and `timeline-panel` styles.

This PR is mainly to ensure that all styles as part of the directive work in both scenarios of use (Custom tags _and_ attributes).

walterg2 commented:
I also see that `travis-ci` is not building as it's forcing an NPM install of `0.10`. I can remove my `package.json` changes so that this isn't a problem.

rpocklin commented:
Ah ok.  Would it make more sense restricting it to elements then?  Otherwise it will be a major version bump.  No worries about `travis-ci`.

walterg2 commented:
I'm not sure it would be a major bump as your code already supports using the timeline via `data` attributes. There were just a few styles (and a couple of places where the styles didn't line up as a result) that used the `timeline-*` element instead of a class. This is all I really changed.

Note: I did have to add the `event.badgeClass` to the templates for those things as well, but it was solely for styling.

walterg2 commented:
Also, removing the ability to use it via attributes _would_, IMHO, be a breaking change and require a major version bump.

walterg2 commented:
Hey @rpocklin. Just wanted to follow up on this change. Is there anything you'd like me to clarify, change or clean up before this is accepted?

rpocklin commented:
Nah you've explained it well and make a good point re: supporting data
elements.  Give me a few days and I'll take some action.

Thanks.

On 3 May 2016 12:46 pm, "George T Walters II" notifications@github.com
wrote:

> Hey @rpocklin. Just wanted to follow up on this change. Is there anything
> you'd like me to clarify, change or clean up before this is accepted?
> 
> —
> You are receiving this because you were mentioned.
> 
> Reply to this email directly or view it on GitHub

rpocklin commented:
Actual styling is ok but the all animation is broken which is a regression.

walterg2 commented:
@rpocklin - Sorry I am just getting to this, but I corrected the issue with the animations not working and updated based on your latest commits to `master` last week. Please let me know if there's anything else you'd like changed in this.

rpocklin commented:
The arrows are messed up (wrong colour).
